apps/openmw/mwmechanics/aicombataction.cpp

```diff
@@ -594,13 +592,17 @@ namespace MWMechanics
                     return 0.f;
             }
         }
-        else
-        {
-            rating *= (effect.mMagnMin + effect.mMagnMax)/2.f;
-        }
 
-        if (!(magicEffect->mData.mFlags & ESM::MagicEffect::NoDuration))
-            rating *= effect.mDuration;
+        rating *= ((effect.mMagnMin + effect.mMagnMax) * (effect.mDuration > 0 ? effect.mDuration : 1) + effect.mArea);
```

Comment:
NoMagnitude spells have magnMin and magnMax = 1. I just took vanilla magicka cost calculation formula from autocalcspell.cpp and spellcasting.cpp. If we do not have a sane default value, we need to fix these files too.